src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -923,6 +923,45 @@ public Integer getNumberOfCorrectionRounds() {
         }
     }
 
+    /** Helper method which does a hard copy of the Grading Criteria
+     *
+     *
+     * @param exerciseImportService@return A clone of the grading criteria list
+     */
+    public List<GradingCriterion> copyGradingCriteria() {
+        List<GradingCriterion> newGradingCriteria = new ArrayList<>();
+        for (GradingCriterion originalGradingCriterion : getGradingCriteria()) {
+            GradingCriterion newGradingCriterion = new GradingCriterion();
+            newGradingCriterion.setExercise(this);
+            newGradingCriterion.setTitle(originalGradingCriterion.getTitle());
+            newGradingCriterion.setStructuredGradingInstructions(copyGradingInstruction(originalGradingCriterion, newGradingCriterion));
+            newGradingCriteria.add(newGradingCriterion);
+        }
+        return newGradingCriteria;
+    }
+
+    /** Helper method which does a hard copy of the Grading Instructions
+     *
+     * @param originalGradingCriterion The original grading criterion which contains the grading instructions
+     * @param newGradingCriterion The cloned grading criterion in which we insert the grading instructions
+     * @return A clone of the grading instruction list of the grading criterion
+     */
+    private List<GradingInstruction> copyGradingInstruction(GradingCriterion originalGradingCriterion, GradingCriterion newGradingCriterion) {
+        List<GradingInstruction> newGradingInstructions = new ArrayList<>();
+        for (GradingInstruction originalGradingInstruction : originalGradingCriterion.getStructuredGradingInstructions()) {
+            GradingInstruction newGradingInstruction = new GradingInstruction();
+            newGradingInstruction.setCredits(originalGradingInstruction.getCredits());
+            newGradingInstruction.setFeedback(originalGradingInstruction.getFeedback());
+            newGradingInstruction.setGradingScale(originalGradingInstruction.getGradingScale());
+            newGradingInstruction.setInstructionDescription(originalGradingInstruction.getInstructionDescription());
+            newGradingInstruction.setUsageCount(originalGradingInstruction.getUsageCount());
+            newGradingInstruction.setGradingCriterion(newGradingCriterion);
```

Comment:
We should probably move that part of the code into a method in GradingCriterion in case we need in at some other place in the future too.